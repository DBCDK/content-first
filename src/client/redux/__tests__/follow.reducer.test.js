import followReducer, {follow, unfollow} from '../follow.reducer';

describe('followReducer', () => {
  test('follow a list', () => {
    let state = followReducer({}, follow('some-id-1', 'list', 0));
    expect(state).toMatchSnapshot();
  });
  test('unfollow a list', () => {
    let state = followReducer({}, follow('some-id-1', 'list', 0));
    state = followReducer(state, follow('some-id-2', 'list', 0));
    expect(state).toMatchSnapshot();
    state = followReducer(state, unfollow('some-id-1', 'follow-id-123', 0));
    expect(state).toMatchSnapshot();
  });
});
