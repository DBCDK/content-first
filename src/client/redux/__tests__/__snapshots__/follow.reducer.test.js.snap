// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`followReducer follow a list 1`] = `
Object {
  "some-id-1": Object {
    "cat": "list",
    "id": "some-id-1",
    "type": "follows",
  },
}
`;

exports[`followReducer unfollow a list 1`] = `
Object {
  "some-id-1": Object {
    "cat": "list",
    "id": "some-id-1",
    "type": "follows",
  },
  "some-id-2": Object {
    "cat": "list",
    "id": "some-id-2",
    "type": "follows",
  },
}
`;

exports[`followReducer unfollow a list 2`] = `
Object {
  "some-id-2": Object {
    "cat": "list",
    "id": "some-id-2",
    "type": "follows",
  },
}
`;
